EyeSeeTea/dhis2-android-capture-app
# PSI - bring solution to apply rules in tei dashboard

### :pushpin: References
* **Issue:** https://app.clickup.com/t/1jfzhwm

###   :gear: branches 
**app**: 
       Origin::feature/psi_bring_fix_tei_dashboard_rules Target: develop-psi
**dhis2-android-SDK**: 
       Origin: feature/bring_last_changes_1_4_3
**dhis2-rule-engine**: 
       Origin: ec2b39f051acd5f1a28b2cf940e83176857f4d76
### :tophat: What is the goal?

Backport solution for "PRs that assign value from DE to TEA doesn't work unless you open the registration form" from 2.5 into 2.4.3

### :memo: How is it being implemented?

- [x] Bring solution from https://github.com/dhis2/dhis2-android-capture-app/pull/2389

### :boom: How can it be tested?

I have created a new program Test tracker in dev.psi-mis.org, there is a positive integer tracker entity attribute and a positive integer data element.

- create a new tei and assign value to test positive integer att
- create a new event and assign value to test positive integer DE
- The attribute in TEI should change, sync event should send the TEI with the expected value in the attribute, exit from tei and enter again should show the expected value in the attribute

### :floppy_disk: Requires DB migration?


### :art: UI changes?
